packages/utils/src/cache/messenger.ts

```diff
@@ -0,0 +1,115 @@
+import { MESSENGER_PREFIX } from './constants.js';
+import { Logger } from './helpers.js';
+
+export type Message = {
+  id: number;
+  key: string;
+  value: string;
+}
+
+export type OnMessageCallback = (message: Message) => void;
+export type OnCompleteCallback = () => void;
+
+type StorageNames = { messagePosts: string };
```

Comment:
Would this offer better type safe guard ? 
```suggestion
type StorageNames = { messagePosts: `${string}[message-posts]` };
```

Reply:
Yes, this is safer, I already updated it.